lib/rules/no-duplicate-case.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Rule to disallow a duplicate case label.
+ * @author Dieter Oberkofler
+ * @copyright 2015 Dieter Oberkofler. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    function isEqual(first, second) {
+        if (first.type === "Literal" && second.type === "Literal" && first.value === second.value) {
+            return true;
+        }
+
+        if (first.type === "Identifier" && second.type === "Identifier" && first.name === second.name) {
+            return true;
+        }
+
+        return false;
+    }
+
+    return {
+
+        "SwitchStatement": function(node) {
```

Comment:
This can be done significantly more efficient by keeping a hash table of case names for each `SwitchStatement` and subscribing to "SwitchCase" nodes to update hash table and check for duplicates.